src/Weather/xctherm/XCThermMapOverlay.cpp

```
@@ -0,0 +1,232 @@
+  const auto utc = GetUtcTimeParts();
+  const int idx = PickAutoTimeIndex(hours, utc.hour, utc.minute);
+  const unsigned hour = idx >= 0 ? hours[unsigned(idx)] : hours[0];
+  return ApplyCachedLayerOverlay(layer_index, hour);
+}
+
+void
+RestoreActiveLayerOverlay() noexcept
+{
+  const auto &settings =
+    CommonInterface::GetComputerSettings().weather.xctherm;
+  const int layer_index = ResolveDisplayLayerIndex(settings);
```

Cppcheck (2.20.0)

[error] Rethrowing current exception with 'throw;', it seems there is no current exception to rethrow. If there is no current exception this calls std

(rethrowNoCurrentException)